File: fixtures/scale/java/large/src/cryptobench/ciphers/gronsfeld/GronsfeldKey.java
package cryptobench.ciphers.gronsfeld;

public class GronsfeldKey {
    private final String digits;

    public GronsfeldKey(String digits) {
        this.digits = digits;
    }

    public int digitAt(int position) {
        return digits.charAt(position % digits.length()) - '0';
    }

    public char keyCharAt(int position) {
        return digits.charAt(position % digits.length());
    }

    public static GronsfeldKey defaultKey() {
        return new GronsfeldKey("31415");
    }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/gronsfeld/GronsfeldKey.scala
package cryptobench.ciphers.gronsfeld

/** Parameters for the gronsfeld cipher. */
final case class GronsfeldKey(digits: String) {
  def digitAt(position: Int): Int = digits.charAt(position % digits.length) - '0'
}

object GronsfeldKey {
  def default(): GronsfeldKey = GronsfeldKey("31415")
}
